### PR TITLE
HOCS-2005: rework Axios error handling

### PR DESCRIPTION
**Pull request by LiamMacP**
When an error is thrown within the express middleware there were a lot of instances of logging the entire Axios object. This isn't useful, and ends up spamming our SIEM.

This change removes this instances and adds a axios error middleware that transposes this error into a `GenericError`.

### CODE REVIEW

File: server/middleware/__tests__/caseType.spec.js [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-const getLogger = require('../../libs/logger');
 
 const headers = '__headers__';
 const req = { params: { type: '__type__' }, user: '__user__' };
 const sendStatus = jest.fn();
 const json = jest.fn();
 const res = { json, locals: {}, sendStatus };
 const next = jest.fn();
-const caseTypes = [{ label: 'Case Type 1', value: 'caseType1' }, { label: 'Case Type 2', value: 'caseType2' }];
+const caseTypes = [ { label: 'Case Type 1', value: 'caseType1' }, { label: 'Case Type 2', value: 'caseType2' } ];
```
Comment (zuzak) [inline server/middleware/__tests__/caseType.spec.js:14]: You've put otherwise unrelated cosmetic changes into the same commit as everything else. Consider splitting them into two commits (you can use `git add -p` for this).

**Comment by zuzak** [comment]
You've put a significant number otherwise unrelated cosmetic changes into the same commit as everything else. Consider splitting them into two commits (you can use `git add -p` for this) so that we can review the substantive changes more easily.

File: server/middleware/request.js [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
 const { isProduction } = require('../config');
 const listService = require('../services/list/');
+const { AxiosError } = require('axios');
+
+function axiosErrorMiddleware(err, _req, _res, next) {
+    if (err instanceof AxiosError) {
+        if (err.response) {
+            return next(new GenericError(err.response.data, err.response.status));
+        } else if (err.request) {
+            return next(new GenericError(`Failed to request following endpoint ${err.config.url}`, 500));
```
Comment (zuzak) [inline server/middleware/request.js]: Will this always be 500? Should we pass through `axiosError.status`?

Comment (zuzak) [reply server/middleware/request.js]: It looks like [AxiosError](https://github.com/axios/axios/blob/bdf493cf8b84eb3e3440e72d5725ba0f138e0451/lib/core/AxiosError.js) is fairly generic and will happen for errors below level 7. If an error occurs because of, say, `ECONNABORTED`, we need to pass that through to the logger.

We previously logged too much, but I suspect now we're logging far too little?

**Comment by LiamMacP** [comment]
> You've put a significant number otherwise unrelated cosmetic changes into the same commit as everything else. Consider splitting them into two commits (you can use `git add -p` for this) so that we can review the substantive changes more easily.

While I would normally agree, this is an IDE setting that formats on every save. We even have a pre-commit lint check that would throw errors and force a correction. I don't think it's reasonable to disable that. 

You can toggle the whitespace diff in the `files changed` tab that will remove all of this for you.

File: server/middleware/__tests__/caseType.spec.js (hunk 1 above)
Comment (LiamMacP) [reply server/middleware/__tests__/caseType.spec.js:14]: https://github.com/UKHomeOffice/hocs-management-ui/pull/327#issuecomment-1301886429

File: server/middleware/request.js (hunk 2 above)
Comment (LiamMacP) [reply server/middleware/request.js]: Status is only available on `AxiosResponse` object which you don't have if its not got that far (https://axios-http.com/docs/handling_errors). 

I do think there is a good call here to add the Axios code which contains `ECONNABORTED`.

Comment (LiamMacP) [reply server/middleware/request.js]: This has been rectified.

File: server/middleware/request.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
 const { v4: uuid } = require('uuid');
 const logger = require('../libs/logger');
-const { ValidationError } = require('../models/error');
+const { ValidationError, GenericError } = require('../models/error');
 const { isProduction } = require('../config');
 const listService = require('../services/list/');
+const { AxiosError } = require('axios');
+
+function axiosErrorMiddleware(err, _req, _res, next) {
```
Comment (robertdeniszczyc2) [inline server/middleware/request.js:8]: We could add tests for this in request.spec.js

Comment (LiamMacP) [reply server/middleware/request.js:8]: This has been completed 🙂

**Review by robertdeniszczyc2 — APPROVED** [review]
